lib/src/package_analysis/shape.dart

```diff
@@ -0,0 +1,115 @@
+import 'package:json_annotation/json_annotation.dart';
+import 'package:meta/meta.dart';
+
+part 'shape.g.dart';
+
+/// A Shape for describing a package.
+@sealed
+@JsonSerializable()
+class PackageShape {
+  final List<LibraryShape> libraries;
+
+  /// All the classes defined in the libraries of this package.
+  final List<ClassShape> classes;
+
+  PackageShape({required this.libraries, required this.classes});
+
+  factory PackageShape.fromJson(Map<String, dynamic> json) =>
+      _$PackageShapeFromJson(json);
+
+  Map<String, dynamic> toJson() => _$PackageShapeToJson(this);
+}
+
+/// A Shape for describing a library, which may not be contained in a single
+/// file.
+@sealed
+@JsonSerializable()
+class LibraryShape {
+  /// Unique identifier for the library represented by this [LibraryShape]
+  /// within the given [PackageShape]. This uses the syntax beginning with
+  /// `package:` which would usually be used for importing this library.
+  ///
+  /// Example:
+  /// ```
+  /// package:my_package/my_library.dart
+  /// ```
+  final String uri;
+
+  final List<PropertyShape> getters;
+  final List<PropertyShape> setters;
+
+  /// The top-level functions exported in this library.
+  final List<MethodShape> functions;
+
+  /// `Set` of [ClassShape.id] elements, where each one corresponds to a class
+  /// exported in this library.
+  Set<int> exportedClasses;
+
+  LibraryShape(
+      {required this.uri,
+      required this.getters,
+      required this.setters,
+      required this.functions,
+      required this.exportedClasses});
```

Comment:
I would personally use trailing comma, but I know others feel differently, do whatever you like :shrug: 

I just wanted to point out that it can also be written as:
```suggestion
  LibraryShape({
    required this.uri,
    required this.getters,
    required this.setters,
    required this.functions,
    required this.exportedClasses,
  });
```